### PR TITLE
Tree-Shaking in React Output Target

### PR DESCRIPTION
**Pull request by drakegens**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally for affected output targets
- [x] Tests (`npm test`) were run locally and passed
- [x] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Tree-Shaking is not working for react-output-target

Issue URL: https://github.com/ionic-team/stencil-ds-output-targets/issues/255

Web Components exports will be split into separate files, enabling tree-shaking

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/react-output-target/package-lock.json
Comment (drakegens) [inline packages/react-output-target/package-lock.json:1]: I should revert this file.

File: packages/react-output-target/src/output-react.ts [hunk 1]
```diff
@@ -19,7 +20,7 @@ export async function reactProxyOutput(
   const filteredComponents = getFilteredComponents(outputTarget.excludeComponents, components);
   const rootDir = config.rootDir as string;
   const pkgData = await readPackageJson(rootDir);
-
+  //here
```
Comment (drakegens) [inline packages/react-output-target/src/output-react.ts:23]: rm

**Comment by drakegens** [comment]
working on fixing/adding tests now.

**Comment by drakegens** [comment]
tests added.

**Comment by drakegens** [comment]
Would love some feedback from @JessicaSachs or @tanner-reits or @thetaPC :)

**Review by tanner-reits — COMMENTED** [review]
Hey @drakegens 👋 

Sorry for the delay here! I've been poking at this as I've had time over the last few weeks. Haven't dived into the code changes too much, but am having some issues confirming the initial treeshaking issue. Testing some builds of the Ionic Framework with this implementation and the existing React output target isn't producing a difference in bundle sizes when analyzing with a tool like the [Rollup bundle analyzer](https://www.npmjs.com/package/rollup-plugin-analyzer). Granted, the Ionic Framework does have some additional build/bundle steps that are influencing the final distributed code. 

I tried to test using the Crayons repo in the Github issue, but there are quite a few issues with getting started in that project (probably me doing something wrong). If you have a better reproduction case (or we can reach out in the Github issue), I'm happy to check things out there.

Aside from that, I was seeing some errors when testing in the Ionic Framework after generating the React proxies and then trying to build the React package. It appears this was just an issue with `import` statements getting generated when `export` statements should have been created in the specified `proxiesFile` for the output target config. But, that's a small issue we can work through if/when we have a good reproduction.

Please let me know if you have questions or want clarification on anything! Hopefully we can get this squared away quickly!

**Comment by drakegens** [comment]
Hey there @tanner-reits!

Sorry for the delay.

I was able to see a ~1mb decrease in our bundle size (an app that consumes web components built with Stencil and wrapped via React Output Target). Unfortunately I can't share my employer's code. So we'll need to find another reproduction case.

I could try and use the Crayons project too.

**Comment by ReneWerner87** [comment]
@drakegens @tanner-reits any progress on this ?

**Comment by christian-bromann** [comment]
Fixed by #432

**Pull request closed without merging**